### PR TITLE
Ensure slots are initialised. Return non-failing connections to pool

### PR DESCRIPTION
**Pull request by dim**
Fixes for #86

### CODE REVIEW

File: cluster.go [hunk 1]
```diff
@@ -176,14 +179,15 @@ func (c *ClusterClient) resetClients() (err error) {
...
 
-	c.slots = make([][]string, hashSlots)
 	c.resetClients()
-
 	seen := make(map[string]struct{})
 	for _, addr := range c.addrs {
 		seen[addr] = struct{}{}
 	}
 
+	for i := 0; i < hashSlots; i++ {
```
Comment (vmihailenco) [inline cluster.go]: This should be much more slower than `make([][]string, hashSlots)`. But I don't insist.

**Comment by vmihailenco** [comment]
LGTM

File: cluster.go (hunk 1 above)
Comment (dim) [inline cluster.go]: You would be surprised:

```go
package x

import "testing"

func BenchmarkMake(b *testing.B) {
	for i := 0; i < b.N; i++ {
		_ = make([][]string, 20000)
	}
}

func BenchmarkUnset(b *testing.B) {
	x := make([][]string, 20000)
	b.ResetTimer()
	for i := 0; i < b.N; i++ {
		for j := 0; j < 20000; j++ {
			x[j] = x[j][:0]
		}
	}
}
```

```bash
PASS
BenchmarkMake	    5000	    234386 ns/op
BenchmarkUnset	   50000	     35669 ns/op
```

Comment (vmihailenco) [inline cluster.go]: Ha, sure I am wrong. `x[j][:0]` just sets slice length, but `make([][]string, 20000)` allocates new slice. No surprise here :)

Comment (dim) [inline cluster.go]: Cool, trying to avoid allocations where possible, old habit :)

**Pull request merged**